src/uct/base/uct_iface.c

```diff
@@ -59,6 +59,70 @@ static ucs_stats_class_t uct_iface_stats_class = {
 };
 #endif
 
+typedef struct {
+    uint64_t flag;
+    char     *name;
+} uct_iface_cap_info_t;
+
+static const uct_iface_cap_info_t uct_iface_cap_infos[] = {
+    {UCT_IFACE_FLAG_AM_SHORT, "am_short"},
+    {UCT_IFACE_FLAG_AM_BCOPY, "am_bcopy"},
+    {UCT_IFACE_FLAG_AM_ZCOPY, "am_zcopy"},
+    {UCT_IFACE_FLAG_PENDING, "pending"},
+    {UCT_IFACE_FLAG_PUT_SHORT, "put_short"},
+    {UCT_IFACE_FLAG_PUT_BCOPY, "put_bcopy"},
+    {UCT_IFACE_FLAG_PUT_ZCOPY, "put_zcopy"},
+    {UCT_IFACE_FLAG_GET_SHORT, "get_short"},
+    {UCT_IFACE_FLAG_GET_BCOPY, "get_bcopy"},
+    {UCT_IFACE_FLAG_GET_ZCOPY, "get_zcopy"},
+    {UCT_IFACE_FLAG_ATOMIC_CPU, "atomic_cpu"},
+    {UCT_IFACE_FLAG_ATOMIC_DEVICE, "atomic_device"},
+    {UCT_IFACE_FLAG_ERRHANDLE_SHORT_BUF, "errhandle_short_buf"},
+    {UCT_IFACE_FLAG_ERRHANDLE_BCOPY_BUF, "errhandle_bcopy_buf"},
+    {UCT_IFACE_FLAG_ERRHANDLE_ZCOPY_BUF, "errhandle_zcopy_buf"},
+    {UCT_IFACE_FLAG_ERRHANDLE_AM_ID, "errhandle_am_id"},
+    {UCT_IFACE_FLAG_ERRHANDLE_REMOTE_MEM, "errhandle_remote_mem"},
+    {UCT_IFACE_FLAG_ERRHANDLE_BCOPY_LEN, "errhandle_bcopy_len"},
+    {UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE, "errhandle_peer_failure"},
+    {UCT_IFACE_FLAG_EP_CHECK, "ep_check"},
+    {UCT_IFACE_FLAG_CONNECT_TO_IFACE, "connect_to_iface"},
+    {UCT_IFACE_FLAG_CONNECT_TO_EP, "connect_to_ep"},
+    {UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR, "connect_to_sockaddr"},
+    {UCT_IFACE_FLAG_AM_DUP, "am_dup"},
+    {UCT_IFACE_FLAG_CB_SYNC, "cb_sync"},
+    {UCT_IFACE_FLAG_CB_ASYNC, "cb_async"},
+    {UCT_IFACE_FLAG_EP_KEEPALIVE, "ep_keepalive"},
+    {UCT_IFACE_FLAG_TAG_EAGER_SHORT, "tag_eager_short"},
+    {UCT_IFACE_FLAG_TAG_EAGER_BCOPY, "tag_eager_bcopy"},
+    {UCT_IFACE_FLAG_TAG_EAGER_ZCOPY, "tag_eager_zcopy"},
+    {UCT_IFACE_FLAG_TAG_RNDV_ZCOPY, "tag_rndv_zcopy"},
+};
+
+#define uct_iface_vfs_show_f(_op, _attr) uct_iface_vfs_show_##_op##_##_attr
+
+#define uct_iface_vfs_show(_op, _attr, _append_to_strb) \
```

Comment:
maybe add user-defined custom argument(s) to show function. to avoid need to define multiple functions

Reply:
The callback function should be parametrized by attribute value to avoid to define multiple functions. Then the attribute value should be stored in VFS node. In other words, iface attribute structure will be stored in VFS node representing UCT iface.

Reply:
can we define a global array of structs with (attribute offset, name, ...), and the callback parameter will be the pointer to such struct? it will query, read from specific byte offset in iface_attr, etc?

Reply:
I'm afraid that the logic will be a little complicated because there are nested structures in `uct_iface_attr`. It won't be hard to get an issue in the logic, e.g. adding new fields to an innermost structure of `uct_iface_attr`.

Reply:
if we calc offset by ucs macro it would be updated even if we add new fields

Reply:
Ok, now I see.
I will update PR after merging of #6713.